Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/DbCommandBuilders/StoredProcedureDbCommandBuilderTest.cs

```diff
@@ -0,0 +1,156 @@
+  {
+    var dataParameterCollectionStrictMock = new Mock<IDataParameterCollection>(MockBehavior.Strict);
+
+    var dbCommandStub = new Mock<IDbCommand>();
+    dbCommandStub.Setup(stub => stub.Parameters).Returns(dataParameterCollectionStrictMock.Object);
+    dbCommandStub.SetupProperty(stub => stub.CommandText);
+
+    var dbCommandFactoryStub = new Mock<IDbCommandFactory>();
+    dbCommandFactoryStub.Setup(stub => stub.CreateDbCommand()).Returns(dbCommandStub.Object);
+
+    var commandBuilder = new StoredProcedureDbCommandBuilder("Procedure", [], GetSqlDialectStub());
+    var result = commandBuilder.Create(dbCommandFactoryStub.Object);
+
+    dataParameterCollectionStrictMock.Verify();
+    _valueParameterDefinitionMock.Verify();
+    Assert.That(result, Is.SameAs(dbCommandStub.Object));
+  }
+
+  [Test]
+  public void Create_TransfersCommandTextToQueryCommand ()
+  {
+    var dataParameterCollectionStrictMock = new Mock<IDataParameterCollection>(MockBehavior.Strict);
+
+    _valueParameterDefinitionMock.Setup(mock => mock.GetParameterValue(5)).Returns(5);
+
+    var dbCommandStub = new Mock<IDbCommand>();
+    dbCommandStub.Setup(stub => stub.Parameters).Returns(dataParameterCollectionStrictMock.Object);
+    dbCommandStub.SetupProperty(stub => stub.CommandText);
+
+    var dbCommandFactoryStub = new Mock<IDbCommandFactory>();
+    dbCommandFactoryStub.Setup(stub => stub.CreateDbCommand()).Returns(dbCommandStub.Object);
+
+    var commandBuilder = new StoredProcedureDbCommandBuilder("Procedure", [], GetSqlDialectStub());
+    var result = commandBuilder.Create(dbCommandFactoryStub.Object);
+
+    dataParameterCollectionStrictMock.Verify();
+    _valueParameterDefinitionMock.Verify();
+    Assert.That(result.CommandText, Is.EqualTo("Procedure"));
+  }
+
+  [Test]
+  public void Create_ThrowsOnTextParameter ()
+  {
+    var dataParameterCollectionStrictMock = new Mock<IDataParameterCollection>(MockBehavior.Strict);
+
+    var dbCommandStub = new Mock<IDbCommand>();
+    dbCommandStub.Setup(stub => stub.Parameters).Returns(dataParameterCollectionStrictMock.Object);
+    dbCommandStub.SetupProperty(stub => stub.CommandText);
+
+    var dbCommandFactoryStub = new Mock<IDbCommandFactory>();
+    dbCommandFactoryStub.Setup(stub => stub.CreateDbCommand()).Returns(dbCommandStub.Object);
+
+    var textQueryParameterDefinitionMock = new Mock<IDataParameterDefinition>(MockBehavior.Strict);
+    var textQueryParameter = new QueryParameter("#textParam", "replacementText", QueryParameterType.Text);
+    var textQueryParameterWithDataDefinition = new QueryParameterWithDataParameterDefinition(textQueryParameter, textQueryParameterDefinitionMock.Object);
+
+    var commandBuilder = new StoredProcedureDbCommandBuilder("Procedure", [ textQueryParameterWithDataDefinition ], GetSqlDialectStub());
+    Assert.Throws<NotSupportedException>(() => commandBuilder.Create(dbCommandFactoryStub.Object));
```

Comment:
Please update to `Assert.That(..., Throws....)` We rely mostly on `Assert.That` these days.